File: BPMOPMobile_VietNamAirLines/BPMOPMobile_VietNamAirLines/BPMOPMobile.Droid.Core/Component/ComponentButtonBot.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Android.App;
using Android.Content;
using Android.Content.Res;
using Android.OS;
using Android.Runtime;
using Android.Util;
using Android.Views;
using Android.Widget;
using BPMOPMobile.Bean;
using BPMOPMobile.Class;
using BPMOPMobile.Droid.Core.Class;
using BPMOPMobile.Droid.Core.Common;
using BPMOPMobile.Droid.Core.Component;

namespace BPMOPMobile.Droid.Core.Component
{
    public class ComponentButtonBot : ComponentBase
    {
        public ViewElement _element { get; set; }
        public LinearLayout _parentView { get; set; }
        public Activity _mainAct;
        public ViewRow _controlData;
        public Resources _resource;
        public ControlButtonBot _btnAction1, _btnAction2, _btnAction3;
        public List<ButtonAction> _lstActionMore = new List<ButtonAction>();
        public ComponentButtonBot(Activity _mainAct, LinearLayout _parentView, ViewRow _controlData, Resources _resource)
        {
            this._mainAct = _mainAct;
            this._parentView = _parentView;
            this._controlData = _controlData;
            this._resource = _resource;

            _controlData.Elements = CmmFunction.SortListElementAction(_controlData.Elements);
            InitializeComponent();
        }
        public override void InitializeComponent()
        {
            if (_controlData.Elements.Any())
                _btnAction1 = new ControlButtonBot(_mainAct, _parentView, _controlData.Elements[0], _resource);
            else
                _btnAction1 = new ControlButtonBot(_mainAct);

            if (_controlData.Elements.Count < 2)
                _btnAction2 = new ControlButtonBot(_mainAct);
            else
                _btnAction2 = new ControlButtonBot(_mainAct, _parentView, _controlData.Elements[1], _resource);

            if (_controlData.Elements.Count < 3)
                _btnAction3 = new ControlButtonBot(_mainAct);
            else
            {
                ViewElement _elementMore = new ViewElement { ID = "more", Value = "more", Hidden = false };
                _btnAction3 = new ControlButtonBot(_mainAct, _parentView, _elementMore, _resource);
                _lstActionMore = new List<ButtonAction>();
                for (int i = 2; i < _controlData.Elements.Count; i++)
                {
                    ButtonAction action = new ButtonAction { ID = Convert.ToInt32(_controlData.Elements[i].ID), Title = _controlData.Elements[i].Title, Value = _controlData.Elements[i].Value };
                    _lstActionMore.Add(action);
                }
            }
        }

        public override void InitializeFrameView(LinearLayout frame)
        {
            base.InitializeFrameView(frame);
            int _height = (int)CmmDroidFunction.ConvertDpToPixel(35, frame.Context);

            LinearLayout _lncontent = new LinearLayout(_mainAct);
            int _paddingLeftRight = (int)TypedValue.ApplyDimension(ComplexUnitType.Dip, 12, _mainAct.Resources.DisplayMetrics);
            _lncontent.LayoutParameters = new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MatchParent, LinearLayout.LayoutParams.MatchParent, 1);
            _lncontent.Orientation = Android.Widget.Orientation.Horizontal;
            _lncontent.SetPadding(_paddingLeftRight, 0, _paddingLeftRight, 0);

            LinearLayout _lnAction1 = new LinearLayout(_mainAct);
            LinearLayout _lnAction2 = new LinearLayout(_mainAct);
            LinearLayout _lnActionMore = new LinearLayout(_mainAct);

            _lnAction1.LayoutParameters = new LinearLayout.LayoutParams(0, _height, 0.5f); // 0.5
            _lnAction2.LayoutParameters = new LinearLayout.LayoutParams(0, _height, 0.5f); // 0.5
            _lnActionMore.LayoutParameters = new LinearLayout.LayoutParams(LinearLayout.LayoutParams.WrapContent, _height);

            _btnAction1.InitializeFrameView(_lnAction1); // Init View vào LinearLayout
            _btnAction2.InitializeFrameView(_lnAction2); // Init View vào LinearLayout
            _btnAction3.InitializeFrameView(_lnActionMore); // Init View vào LinearLayout

            _lncontent.AddView(_lnAction1);
            _lncontent.AddView(_lnAction2);
            _lncontent.AddView(_lnActionMore);
            frame.AddView(_lncontent);
        }
        public override void SetTitle()
        {
            _btnAction1.SetTitle();
            _btnAction2.SetTitle();
            _btnAction3.SetTitle();
        }
        public override void SetValue()
        {
            base.SetValue();

            _btnAction1.SetValue();
            _btnAction2.SetValue();
            _btnAction3.SetValue();
        }
        public override void SetProprety()
        {
            _btnAction1.SetProprety();
            _btnAction2.SetProprety();
            _btnAction3.SetProprety();
        }
        public override void SetEnable()
        {
            base.SetEnable();

            _btnAction1.SetEnable();
            _btnAction2.SetEnable();
            _btnAction3.SetProprety();
        }
    }
}

File: VNABPMOP/app/src/main/java/com/vuthao/bpmop/core/component/ComponentButtonBot.java
package com.vuthao.bpmop.core.component;

import android.app.Activity;
import android.content.res.Resources;
import android.util.TypedValue;
import android.widget.LinearLayout;

import com.vuthao.bpmop.base.Functions;
import com.vuthao.bpmop.base.model.custom.ButtonAction;
import com.vuthao.bpmop.base.model.dynamic.ViewElement;
import com.vuthao.bpmop.base.model.dynamic.ViewRow;
import com.vuthao.bpmop.detail.custom.DetailFunc;

import java.util.ArrayList;

public class ComponentButtonBot extends ComponentBase {
    private ViewElement element;
    private final LinearLayout parentView;
    private final Activity mainAct;
    private final ViewRow controlData;
    private final Resources resource;
    private ControlButtonBot btnAction1, btnAction2, btnAction3;
    private ArrayList<ButtonAction> lstActionMore = new ArrayList<>();

    public ComponentButtonBot(Activity mainAct, LinearLayout parentView, ViewRow controlData, Resources resource) {
        this.mainAct = mainAct;
        this.parentView = parentView;
        this.controlData = controlData;
        this.resource = resource;

        this.controlData.setElements(DetailFunc.share.sortListElementAction(this.controlData.getElements()));
        initializeComponent();
    }

    public ArrayList<ButtonAction> getLstActionMore() {
        return lstActionMore;
    }

    public void setLstActionMore(ArrayList<ButtonAction> lstActionMore) {
        this.lstActionMore = lstActionMore;
    }

    @Override
    public void initializeComponent() {
        if (controlData.getElements().size() > 0) {
            btnAction1 = new ControlButtonBot(mainAct, parentView, controlData.getElements().get(0), resource);
        }

        if (controlData.getElements().size() > 1) {
            btnAction2 = new ControlButtonBot(mainAct, parentView, controlData.getElements().get(1), resource);
        }

        if (controlData.getElements().size() > 2) {
            ViewElement elementMore = new ViewElement();
            elementMore.setID("more");
            elementMore.setValue("more");
            elementMore.setHidden(false);

            btnAction3 = new ControlButtonBot(mainAct, parentView, elementMore, resource);
            lstActionMore = new ArrayList<>();
            for (int i = 2; i < controlData.getElements().size(); i++) {
                ButtonAction action = new ButtonAction();
                action.setID(Integer.parseInt(controlData.getElements().get(i).getID()));
                action.setTitle(controlData.getElements().get(i).getTitle());
                action.setValue(controlData.getElements().get(i).getValue());

                lstActionMore.add(action);
            }
        }
    }

    @Override
    public void initializeFrameView(LinearLayout frame) {
        super.initializeFrameView(frame);

        int height = Functions.share.convertDpToPixel(35, frame.getContext());
        LinearLayout lncontent = new LinearLayout(mainAct);
        int paddingLeftRight = (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, 12, mainAct.getResources().getDisplayMetrics());
        lncontent.setLayoutParams(new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.MATCH_PARENT, 1));
        lncontent.setOrientation(LinearLayout.HORIZONTAL);
        lncontent.setPadding(paddingLeftRight, 0, paddingLeftRight, 0);

        LinearLayout lnAction1 = new LinearLayout(mainAct);
        LinearLayout lnAction2 = new LinearLayout(mainAct);
        LinearLayout lnActionMore = new LinearLayout(mainAct);

        lnAction1.setLayoutParams(new LinearLayout.LayoutParams(0, height, 0.5f));
        lnAction2.setLayoutParams(new LinearLayout.LayoutParams(0, height, 0.5f));
        lnActionMore.setLayoutParams(new LinearLayout.LayoutParams(LinearLayout.LayoutParams.WRAP_CONTENT, height));

        if (btnAction1 != null) {
            btnAction1.initializeFrameView(lnAction1);
        }

        if (btnAction2 != null) {
            btnAction2.initializeFrameView(lnAction2);
        }

        if (btnAction3 != null) {
            btnAction3.initializeFrameView(lnActionMore);
        }

        lncontent.addView(lnAction1);
        lncontent.addView(lnAction2);
        lncontent.addView(lnActionMore);
        frame.addView(lncontent);
    }

    @Override
    public void setTitle() {
        if (btnAction1 != null) {
            btnAction1.setTitle();
        }
        if (btnAction2 != null) {
            btnAction2.setTitle();
        }
        if (btnAction3 != null) {
            btnAction3.setTitle();
        }
    }

    @Override
    public void setValue() {
        super.setValue();

        if (btnAction1 != null) {
            btnAction1.setValue();
        }
        if (btnAction2 != null) {
            btnAction2.setValue();
        }
        if (btnAction3 != null) {
            btnAction3.setValue();
        }
    }

    @Override
    public void setProprety() {
        if (btnAction1 != null) {
            btnAction1.setProprety();
        }
        if (btnAction2 != null) {
            btnAction2.setProprety();
        }
        if (btnAction3 != null) {
            btnAction3.setProprety();
        }
    }

    @Override
    public void setEnable() {
        if (btnAction1 != null) {
            btnAction1.setEnable();
        }
        if (btnAction2 != null) {
            btnAction2.setEnable();
        }
        if (btnAction3 != null) {
            btnAction3.setEnable();
        }
    }
}
